jeremywiebe/harp
# Web 2067

There were some NPM vulnerabilities shown in the documentation-theme project which was then traced back to here. Thus, some dependencies were added and/or updated.

 **JIRA**: https://mobify.atlassian.net/secure/RapidBoard.jspa?rapidView=271&projectKey=WEB&modal=detail&selectedIssue=WEB-2067

## Changes
- Pulled upstream from https://github.com/sintaxi/harp
- Address some other NPM vulnerabilities with `npm audit` and following the prompts 

## How to test-drive this PR
- ~~I am honestly not sure how this can be tested~~
- Change npm version to 6.0.0 via `npm install -g npm@6.0.0`
- Run `npm audit`
- See that there are the following vulnerabilities:
[!] 12 vulnerabilities found - Packages audited: 4247 (149 dev, 195 optional)
    Severity: 2 low | 9 moderate | 1 high

- The high vulnerabilities results from our terraform as seen through running `npm list marked`:
harp@0.29.0 /Users/lansichu/harp
├── marked@0.5.1 
└─┬ terraform@1.5.0 (github:mobify/terraform#41d33a88eacc895e9e99e275a32ba43373b5e537)
  └── marked@0.3.6 

- I am hoping that this will be solved with the dependency updates/installations with terraform
- From what I know, this project is a dependency for another project (documentation-theme) and that project is being used in three other projects as dependencies (one of them being the progressive-web-SDK)
